### PR TITLE
Ubuntu phone

### PR DESCRIPTION
**Pull request by renatofilho**
Some small fixes and a few behaviors changes used by ubuntu-phone.

### CODE REVIEW

File: libbuteosyncfw/common/NetworkManager.cpp [hunk 1]
```diff
@@ -110,12 +142,54 @@ void NetworkManager::sessionConnectionTimeout()
...
+            }
+        }
+    }
+    LOG_DEBUG("New state:" << isOnline << " New type: " << bearerTypeName << "(" << connectionType << ")");
+    if ((isOnline != m_isOnline) ||
+        (Sync::InternetConnectionType(connectionType) != m_connectionType))
+    {
+        m_isOnline = isOnline;
+        m_connectionType =  Sync::InternetConnectionType(connectionType);
+        emit statusChanged(m_isOnline, m_connectionType);
```
Comment (chriadam) [inline libbuteosyncfw/common/NetworkManager.cpp]: the old slotOnlineStateChanged() used to emit valueChanged() instead of statusChanged() - is this an intentional change?

File: libbuteosyncfw/common/TransportTracker.cpp [hunk 2]
```diff
@@ -175,19 +177,17 @@ void TransportTracker::updateState(Sync::ConnectivityType aType,
     }
     if(oldState != aState)
     {
-        if (aType != Sync::CONNECTIVITY_INTERNET) {
+        if (aType != Sync::CONNECTIVITY_INTERNET)
+        {
             emit connectivityStateChanged(aType, aState);
         }
-        else {
```
Comment (chriadam) [inline libbuteosyncfw/common/TransportTracker.cpp]: why remove this block?

File: libbuteosyncfw/profile/ProfileManager.cpp [hunk 3]
```diff
@@ -739,6 +738,10 @@ QString ProfileManager::updateProfile(const Profile &aProfile)
     bool exists = d_ptr->profileExists(aProfile.name(),aProfile.type());
 
     QString profileId("");
+    if(d_ptr->save(aProfile)) {
```
Comment (chriadam) [inline libbuteosyncfw/profile/ProfileManager.cpp]: why was this moved?  Can a comment be added here?

File: msyncd/AccountsHelper.cpp [hunk 4]
```diff
@@ -454,6 +457,15 @@ void AccountsHelper::registerAccountListener(Accounts::AccountId id)
     QObject::connect(account, SIGNAL(enabledChanged(const QString&, bool)),
                      this, SLOT(slotAccountEnabledChanged(const QString&, bool)));
 
+    // Account SOC
```
Comment (chriadam) [inline msyncd/AccountsHelper.cpp]: better to expand this to SyncOnChange instead of just SOC

File: msyncd/bin/msyncd.conf [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+stop on runlevel [06]
+
+# give some time to nm be ready
+pre-start script
+    sleep 10;
+end script
+
+respawn
+
+env MSYNCD_LOGGING_LEVEL=10
```
Comment (chriadam) [inline msyncd/bin/msyncd.conf]: not sure you want that level of logging in production environment

File: msyncd/bin/msyncd.conf [hunk 6]
```diff
@@ -0,0 +1,15 @@
+description "address-book-service"
```
Comment (chriadam) [inline msyncd/bin/msyncd.conf]: Description probably a copy/paste error

File: msyncd/msyncd-app.pro [hunk 7]
```diff
@@ -59,13 +59,16 @@ target.path = /usr/bin/
...
 service.path = /usr/lib/systemd/user/
+upstart.path = /usr/share/upstart/sessions/
 syncwidget.path = /etc/syncwidget/
 syncwidget.files = com.meego.msyncd
 INSTALLS += target \
     loglevel \
     syncwidget \
-    service
+    service \
+    upstart
```
Comment (chriadam) [inline msyncd/msyncd-app.pro]: Personally, I'd prefer upstart-related stuff to be installed by the debian install scripts instead of the .pro INSTALLS section...

File: msyncd/synchronizer.cpp [hunk 8]
```diff
@@ -115,8 +115,9 @@ bool Synchronizer::initialize()
             this, SLOT(slotSyncStatus(QString, int, QString, int)),
             Qt::QueuedConnection);
 
+    // use queued connection because the profile will be stored after the signal
     connect(&iProfileManager ,SIGNAL(signalProfileChanged(QString,int,QString)),
-            this, SIGNAL(signalProfileChanged(QString,int,QString)));
+            this, SLOT(slotProfileChanged(QString,int,QString)), Qt::QueuedConnection);
```
Comment (chriadam) [inline msyncd/synchronizer.cpp]: you've changed this to the slot instead of the signal - I assume this was just a bug in the original code, and not a trampoline for the Out Of Process plugins?

File: msyncd/synchronizer.cpp [hunk 9]
```diff
@@ -184,7 +186,7 @@ bool Synchronizer::initialize()
         else
         {
             QObject::connect(&iSyncOnChangeScheduler, SIGNAL(syncNow(QString)),
-                             this, SLOT(startSync(QString)),
+                             this, SLOT(startScheduledSync(QString)),
```
Comment (chriadam) [inline msyncd/synchronizer.cpp]: In this case, I don't think it's a scheduled sync (ie, not a sync caused by the sync schedule, but instead it's a sync caused by an account change).  So, I think it should be startSync() instead of startScheduledSync().  Do you agree, or am I misunderstanding the semantics?

File: msyncd/synchronizer.cpp [hunk 10]
```diff
@@ -217,7 +219,7 @@ void Synchronizer::enableSOCSlot(const QString& aProfileName)
         else
         {
             QObject::connect(&iSyncOnChangeScheduler, SIGNAL(syncNow(const QString&)),
-                             this, SLOT(startSync(const QString&)),
+                             this, SLOT(startScheduledSync(const QString&)),
```
Comment (chriadam) [inline msyncd/synchronizer.cpp]: ditto to above comment

File: msyncd/synchronizer.cpp [hunk 11]
```diff
@@ -291,55 +293,27 @@ bool Synchronizer::startScheduledSync(QString aProfileName)
...
-        QObject::connect(iNetworkManager, SIGNAL(connectionSuccess()), this,
-                    SLOT(slotNetworkSessionOpened()), Qt::QueuedConnection);
-        QObject::connect(iNetworkManager, SIGNAL(connectionError()), this,
-                    SLOT(slotNetworkSessionError()), Qt::QueuedConnection);
-        // Request for a network session
-        iNetworkManager->connectSession(true);
+         LOG_DEBUG("Wait for internet connection:" << aProfileName);
+         if (iNetworkManager->isOnline())
+         {
+             LOG_DEBUG("Connection over mobile data plan. The sync will be postponed untill a full connection is available;");
```
Comment (chriadam) [inline msyncd/synchronizer.cpp]: This should not be hard-coded into the daemon, but instead be configurable (perhaps via a dconf key or similar).

File: msyncd/synchronizer.cpp [hunk 12]
```diff
@@ -1903,10 +1900,22 @@ QStringList Synchronizer::syncProfilesByType(const QString &aType)
...
-    if(!aState) {
+    LOG_DEBUG("Network state changed: OnLine:" << aState << " connection type:" <<  type);
+    if (acceptScheduledSync(aState, type))
+    {
+        LOG_DEBUG("Restart sync for profiles that need network");
+        QStringList profiles(iWaitingOnlineSyncs);
+        iWaitingOnlineSyncs.clear();
+        foreach(QString profileName, profiles)
+        {
+            startSync(profileName, true);
```
Comment (chriadam) [inline msyncd/synchronizer.cpp]: Are all waiting online syncs also scheduled syncs?  If so, should call startScheduledSync() here, I guess?

File: msyncd/synchronizer.cpp [hunk 13]
```diff
@@ -2069,6 +2078,18 @@ void Synchronizer::removeExternalSyncStatus(const SyncProfile *aProfile)
     }
 }
 
+bool Synchronizer::acceptScheduledSync(bool aConnected, Sync::InternetConnectionType aType) const
+{
+    static QList<Sync::InternetConnectionType> allowedTypes;
```
Comment (chriadam) [inline msyncd/synchronizer.cpp]: again, this should be configurable and not hardcoded.

File: libbuteosyncfw/common/NetworkManager.h [hunk 14]
```diff
@@ -84,7 +92,7 @@ namespace Buteo
              *
              * @param aConnected If true, the device is online
              */
-            void valueChanged(bool aConnected);
+            void statusChanged(bool aConnected, Sync::InternetConnectionType aType);
```
Comment (chriadam) [inline libbuteosyncfw/common/NetworkManager.h]: Somewhat unrelated to this change, but the ClientPlugin::connectivityStateChanged() slot should also receive this information if possible (as that would allow client plugin to decide whether or not to trigger Avatar downloading, depending on some policy -- eg, only download Avatars if WiFi connectivity is available, otherwise, only sync XML data).

**Comment by chriadam** [comment]
@VDVsx can you review this also please?
@lpotter can you check the QNetworkManager related bits?  Will this work with our stack?

File: libbuteosyncfw/common/TransportTracker.cpp (hunk 2 above)
Comment (renatofilho) [inline libbuteosyncfw/common/TransportTracker.cpp]: The  "emit networkStateChanged" is called from "onInternetStateChanged" because it has different arguments.

File: libbuteosyncfw/profile/ProfileManager.cpp (hunk 3 above)
Comment (renatofilho) [inline libbuteosyncfw/profile/ProfileManager.cpp]: // We need to save before emit the signalProfileChanged, if this is the first
    // update the profile will only exists on disk after the save and any operation
    // using this profile triggered by the signal will fail.

File: msyncd/synchronizer.cpp (hunk 9 above)
Comment (renatofilho) [inline msyncd/synchronizer.cpp]: I consider a ScheduledSync any sync that starts without a manually request. The only difference btw a startSync and a startScheduledSync is that the startScheduledSync will check for network connections before call 'startSync'. And I believe that syncs triggered by database change should have the same policy as schedule syncs.

File: msyncd/synchronizer.cpp (hunk 12 above)
Comment (renatofilho) [inline msyncd/synchronizer.cpp]: Yes all 'waiting online' are scheduled syncs, manually sync triggered while offline will return a network error. And as mentioned above the 'startScheduledSync' only checks for network connection before call 'startSync' and since we already knows that a connection is available for sync there is no need to go through the check again.

File: libbuteosyncfw/profile/SyncProfile.cpp [hunk 15]
```diff
@@ -259,15 +259,11 @@ QDateTime SyncProfile::lastSuccessfulSyncTime () const
     if (d_ptr->iLog)
     {
         QList<const SyncResults*> allResults = d_ptr->iLog->allResults();
-        if (allResults.size () > 0)
-        {
-            lastSuccessSyncTime = allResults.at (0)->syncTime ();
```
Comment (VDVsx) [inline libbuteosyncfw/profile/SyncProfile.cpp]: Why is this one removed ?

File: msyncd/AccountsHelper.cpp [hunk 16]
```diff
@@ -186,7 +189,7 @@ void AccountsHelper::slotAccountEnabledChanged(const QString &serviceName, bool
             foreach(SyncProfile *profile, profiles)
             {
                 // See if the service name matches
-                if(serviceName == profile->name())
+                if(serviceName == profile->key(REMOTE_SERVICE_NAME))
                 {
```
Comment (VDVsx) [inline msyncd/AccountsHelper.cpp]: Why this change ? if the profile exit already created when account is loaded above, there's no guarantee that will be the REMOTE_SERVICE_NAME as far as I can see.

File: libbuteosyncfw/profile/SyncProfile.cpp (hunk 15 above)
Comment (renatofilho) [inline libbuteosyncfw/profile/SyncProfile.cpp]: If the log has only one line ant this line is a fail sync. It will consider it as a successful sync.

File: msyncd/AccountsHelper.cpp (hunk 16 above)
Comment (renatofilho) [inline msyncd/AccountsHelper.cpp]: In case a specific service not the "global" account get enabled we need to check which profiles are related to this service and enable/disable it. The profile name is not the same as the service name. For example my profiles created by the account creation has names like "727230507.xml". The only profile that I have with the same name as the service is the "template" profile used to create new ones. As you can see in the function "createProfileForAccount" it will use the profile as template and create a new one we a new name. 
I do not see the case where the profile-name is the same as the service name.

**Comment by chriadam** [comment]
@blammit @pvuorela @hmallat  We need to get this PR into mainline, and then consider what changes we need to put on top to make it work with our stack (e.g., conf file change to allow sync over mobile network / cellular data, perhaps?)

Any other concerns etc?  I guess this will be aimed at 1.2.1 release for us, so should wait until after 1.2.0 branch date to merge to mainline - that gives us at least two weeks to find issues with this PR and push it forward after that.

File: libbuteosyncfw/common/NetworkManager.cpp [hunk 17]
```diff
@@ -110,12 +142,54 @@ void NetworkManager::sessionConnectionTimeout()
...
+    QList<QNetworkConfiguration> activeConfigs = m_networkConfigManager->allConfigurations(QNetworkConfiguration::Active);
+    QNetworkConfiguration::BearerType connectionType = QNetworkConfiguration::BearerUnknown;
+    QString bearerTypeName;
+
+    bool isOnline = activeConfigs.size() > 0;
+    if (isOnline)
+    {
+        // FIXME: due this bug #1444162 on nm the QNetworkConfigurationManager
+        // returns the wrong default connection.
+        // We will consider the connection with the smallest bearType as the
```
Comment (chriadam) [inline libbuteosyncfw/common/NetworkManager.cpp]: bearerType misspelled in comment

File: libbuteosyncfw/common/NetworkManager.cpp [hunk 18]
```diff
@@ -110,12 +142,54 @@ void NetworkManager::sessionConnectionTimeout()
...
+        // default connection, with that wifi and ethernet will be the first one
```
Comment (chriadam) [inline libbuteosyncfw/common/NetworkManager.cpp]: is this always true for every configuration, or an Ubuntu-specific assumption?  Lorn can you comment on this too?

File: libbuteosyncfw/common/NetworkManager.cpp [hunk 19]
```diff
@@ -110,12 +142,54 @@ void NetworkManager::sessionConnectionTimeout()
...
+            if (conf.bearerType() < connectionType)
+            {
+                connectionType = conf.bearerType();
+                bearerTypeName = conf.bearerTypeName();
+            }
+        }
+    }
+    LOG_DEBUG("New state:" << isOnline << " New type: " << bearerTypeName << "(" << connectionType << ")");
+    if ((isOnline != m_isOnline) ||
+        (Sync::InternetConnectionType(connectionType) != m_connectionType))
```
Comment (chriadam) [inline libbuteosyncfw/common/NetworkManager.cpp]: should we prefer static_cast to C-style cast, or are C-style casts more consistent with rest of buteo code?  Are the Sync::InternetConnectionType values directly castable to QNetworkConfiguration::BearerType values, or should we have a conversion function?

File: libbuteosyncfw/profile/ProfileManager.cpp [hunk 20]
```diff
@@ -1254,7 +1258,7 @@ QDateTime ProfileManager::getNextRetryInterval(const SyncProfile* aProfile)
        !iSyncRetriesInfo[aProfile->name()].isEmpty())
     {
        quint32 mins = iSyncRetriesInfo[aProfile->name()].takeFirst();
-       nextRetryInterval = QDateTime::currentDateTime().addSecs(mins * 60); 
+       nextRetryInterval = QDateTime::currentDateTime().addSecs(mins * 60);
```
Comment (chriadam) [inline libbuteosyncfw/profile/ProfileManager.cpp]: Not related to this patch per se, but since this doesn't use UTC timestamps, I wonder if it's susceptible to problems if the device's timezone settings change.  I guess it would just be a missed-sync-trigger, so not a huge issue.

File: libbuteosyncfw/profile/SyncProfile.cpp [hunk 21]
```diff
@@ -259,15 +259,11 @@ QDateTime SyncProfile::lastSuccessfulSyncTime () const
...
-            foreach (const SyncResults* result, allResults) {
-                if ((result->majorCode () == SyncResults::SYNC_RESULT_SUCCESS) &&
-                    (result->minorCode () == SyncResults::NO_ERROR) &&
-                    (result->syncTime () > lastSuccessSyncTime))
-                        lastSuccessSyncTime = result->syncTime ();
-            }
+        foreach (const SyncResults* result, allResults) {
+            if ((result->majorCode () == SyncResults::SYNC_RESULT_SUCCESS) &&
+                (result->minorCode () == SyncResults::NO_ERROR) &&
+                (lastSyncTime().isNull() || (result->syncTime () > lastSuccessSyncTime)))
```
Comment (chriadam) [inline libbuteosyncfw/profile/SyncProfile.cpp]: prefer braces around single-expressions if the condition covers multiple lines, for clarity.

File: msyncd/AccountsHelper.cpp [hunk 22]
```diff
@@ -42,7 +44,8 @@ AccountsHelper::AccountsHelper(ProfileManager &aProfileManager, QObject *aParent
     QObject::connect(iAccountManager, SIGNAL(accountUpdated(Accounts::AccountId)),
                      this, SLOT(slotAccountUpdated(Accounts::AccountId)));
 
-    registerAccountListeners();
+    // load accounts after return from contructor
+    QTimer::singleShot(0, this, SLOT(registerAccountListeners()));
```
Comment (chriadam) [inline msyncd/AccountsHelper.cpp]: could just use QMetaObject::invokeMethod(this, "registerAccountListeners", Qt::QueuedConnection); I guess?

File: msyncd/AccountsHelper.cpp (hunk 16 above)
Comment (chriadam) [inline msyncd/AccountsHelper.cpp]: I agree - but why was this if-check there originally?  Who added it, and what did they state was the reason for doing so?  Or is it legacy from harmattan-era with no purpose?

File: msyncd/synchronizer.cpp (hunk 9 above)
Comment (chriadam) [inline msyncd/synchronizer.cpp]: I'm not certain that that's true.  I think it's worth having semantic separation in the API.

But for now, let's just go with this.  We can revisit later if we decide that there is indeed value in being able to handle sync differently based on the reason for the trigger.

File: msyncd/synchronizer.cpp (hunk 11 above)
Comment (chriadam) [inline msyncd/synchronizer.cpp]: Renato, are you willing to modify the patch to read these sorts of "configuration settings" from dconf keys or a .ini or something?

File: msyncd/synchronizer.cpp (hunk 12 above)
Comment (chriadam) [inline msyncd/synchronizer.cpp]: Good point.  Can you add a comment to that effect here, so that I don't have to remember it?

File: msyncd/AccountsHelper.cpp [hunk 23]
```diff
@@ -42,7 +44,8 @@ AccountsHelper::AccountsHelper(ProfileManager &aProfileManager, QObject *aParent
     QObject::connect(iAccountManager, SIGNAL(accountUpdated(Accounts::AccountId)),
                      this, SLOT(slotAccountUpdated(Accounts::AccountId)));
 
-    registerAccountListeners();
+    // load accounts after return from contructor
```
Comment (pvuorela) [inline msyncd/AccountsHelper.cpp]: Nitpick: not actually true (not after ctor, but when returning to event loop), also more important would be the reason.

File: libbuteosyncfw/common/NetworkManager.cpp [hunk 24]
```diff
@@ -110,12 +142,54 @@ void NetworkManager::sessionConnectionTimeout()
...
+{
+    FUNCTION_CALL_TRACE;
+    QList<QNetworkConfiguration> activeConfigs = m_networkConfigManager->allConfigurations(QNetworkConfiguration::Active);
+    QNetworkConfiguration::BearerType connectionType = QNetworkConfiguration::BearerUnknown;
+    QString bearerTypeName;
+
+    bool isOnline = activeConfigs.size() > 0;
+    if (isOnline)
+    {
+        // FIXME: due this bug #1444162 on nm the QNetworkConfigurationManager
```
Comment (pvuorela) [inline libbuteosyncfw/common/NetworkManager.cpp]: Could have some hint of bug tracker on reference, plain number doesn't tell much. Also, launchpad.net bug is already fixed?

**Comment by renatofilho** [comment]
Hey guys, the packaging guys is asking me if is possible to remove the "debian" dir from the mainstream project (since it will be different for each dist).

**Comment by chriadam** [comment]
Yes. We're using the rpm dir to generate our packages, though, so if your packaging guys want to remove that as well, then it might take us a couple of days to do something about that (e.g., create an "internal" repo which just contains the mainline one as a submodule + the .spec files, or something like that)

**Comment by pvuorela** [comment]
Debian is just legacy here. I suppose existence of rpm/ doesn't do much harm, though.

File: msyncd/msyncd-app.pro [hunk 25]
```diff
@@ -59,13 +59,19 @@ target.path = /usr/bin/
 loglevel.files = bin/set_sync_log_level
 loglevel.path = /etc/buteo/
 service.files = bin/msyncd.service
+upstart.files = bin/msyncd.conf
 service.path = /usr/lib/systemd/user/
+upstart.path = /usr/share/upstart/sessions/
```
Comment (pvuorela) [inline msyncd/msyncd-app.pro]: Upstart lines should probably be next to each other like all the other targets were.

Also if added here, we need to %exclude on rpm. Could maybe have CONFIG option to include this, or even move together with separate debian packaging?

File: libbuteosyncfw/common/NetworkManager.cpp (hunk 24 above)
Comment (renatofilho) [inline libbuteosyncfw/common/NetworkManager.cpp]: Is fixed on wily network-manager (I can remove this workaround already). Did you guys verify if you have the same problem?

File: msyncd/AccountsHelper.cpp (hunk 23 above)
Comment (renatofilho) [inline msyncd/AccountsHelper.cpp]: The signal "enableSOC" is fired on "void AccountsHelper::registerAccountListener(Accounts::AccountId id)" and we need to connect to this signal.

File: libbuteosyncfw/common/Logger.cpp [hunk 26]
```diff
@@ -206,7 +206,7 @@ void Logger::write(int aLevel, const char *aMsg)
     {
         if(aLevel >= QtCriticalMsg)
         {
-            syslog(LOG_CRIT, sysLogMsg.toLocal8Bit().data());
+            syslog(LOG_CRIT, sysLogMsg.toLocal8Bit().data(), "");
```
Comment (pvuorela) [inline libbuteosyncfw/common/Logger.cpp]: This appears kind of bypassing the security warning. Maybe rather syslog(LOG_CRIT, "%s", syslogMsg...)?

File: msyncd/synchronizer.cpp (hunk 11 above)
Comment (renatofilho) [inline msyncd/synchronizer.cpp]: I have a patch already I will push it soon.

File: msyncd/gschemas/com.meego.msyncd.gschema.xml [hunk 27]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<schemalist>
+  <schema id="com.meego.msyncd" path="/com/meego/msyncd/">
+    <key name="allow-scheduled-sync-over-gsm" type="b">
+      <summary>Scheudled Sync over GSM</summary>
```
Comment (pvuorela) [inline msyncd/gschemas/com.meego.msyncd.gschema.xml]: - Typo
- Might be better to use word "cellular" instead of gsm unless you want to control sync over one type of 2G connection.
- Not sure if dependency to QGSettings is good here. Any option to use mlite/MGConfItem which nowadays wraps dconf, or then dconf directly? What's the upstream for QGSettings anyway?

Comment (renatofilho) [inline msyncd/gschemas/com.meego.msyncd.gschema.xml]: Hey sorry I did not realize that QGSettings was a Ubuntu project (https://code.launchpad.net/~system-settings-touch/gsettings-qt/trunk) very similar to MGConfItem.

Probably the best here is use native gconf glib API, to avoid  extra dep.. What do you think?

File: libbuteosyncfw/profile/SyncProfile.cpp (hunk 21 above)
Comment (renatofilho) [inline libbuteosyncfw/profile/SyncProfile.cpp]: fixed

File: msyncd/gschemas/com.meego.msyncd.gschema.xml (hunk 27 above)
Comment (renatofilho) [inline msyncd/gschemas/com.meego.msyncd.gschema.xml]: typo fixed
gsm word replaced.

File: libbuteosyncfw/common/NetworkManager.cpp (hunk 19 above)
Comment (renatofilho) [inline libbuteosyncfw/common/NetworkManager.cpp]: Used C-style cast.
I set individuals enums values to make sure that the enums has the same value as QNetworkConfiguration::BearerType.

File: libbuteosyncfw/common/NetworkManager.cpp (hunk 17 above)
Comment (renatofilho) [inline libbuteosyncfw/common/NetworkManager.cpp]: fixed

**Comment by pvuorela** [comment]
> Probably the best here is use native gconf glib API, to avoid extra dep.. What do you think?

MGConf actually uses dconf nowadays despite the name. GConf is deprecated and shouldn't add use for it.

I understand configuration came from Chris' request not to hard-code network connection behavior. If so and ok with him, could be an option too just having CONFIG option together with some #ifdef instead of trying to agree on configuration mechanism. If dynamic configuration is needed, it can be added separately.

**Comment by chriadam** [comment]
Can we use a QSettings .ini file in a standard location, instead, perhaps?

At the end of the day, I have no strong reason to want it runtime configurable, so if the ifdef is simpler, can do that too.

**Comment by lpotter** [comment]
As far as Network code, quick glance it looks ok.

Pedantic style comment: I see a mixture of bracket style being added/changed.

foo) {

and

foo)
{

As well as a mixture of if( and if (

I personally prefer Qt style of 
foo) {
 if (

It might be best to at least keep is consistent throughout this code.

**Comment by chriadam** [comment]
What remains blocking this PR?  Are there multiple points outstanding, or just the configuration settings stuff?  Have we decided on how we're going to do that?  QSettings or dconf or #ifdef ?

**Comment by renatofilho** [comment]
QSettings replaced by GSettings;
@lpotter I am not sure which one is the best here, most of the code is using (but still a few places using a different style):
if ()
{
}
else
{
}
I think keeping this code style is the best here. (I did not change yes, I will wait for a decision here)

**Comment by renatofilho** [comment]
@chriadam about the commit with the upstart changes we can remove it from this MR. The same for the debian changes if you agree with removing the 'debian' dir from the upstream project.

**Comment by renatofilho** [comment]
@all next week we will release a package of buteo for ubuntu (wily) will be nice if at least part of this MR was merged upstream, to keep the debian/ubuntu patches small as possible.
Thanks

**Comment by chriadam** [comment]
Don't bother changing the style etc in this MR.  I need to do a "fix style" commit which touches all of the buteo components to make them more readable and maintainable and consistent, anyway.

**Comment by chriadam** [comment]
@renatofilho please do remove the upstart changes if possible, and yes as part of this MR feel free to rm the debian dir.

**Comment by chriadam** [comment]
Once those things are done, if nobody has any other concerns, I will merge this one.  Pekka, do you agree, or have I overlooked something important?

File: msyncd/synchronizer.h [hunk 28]
```diff
@@ -446,6 +455,7 @@ private slots:
 #endif
 
     QDBusInterface *iSyncUIInterface;
+    _GSettings *iSettings;
```
Comment (pvuorela) [inline msyncd/synchronizer.h]: Why not plain GSettings?

**Comment by pvuorela** [comment]
Ok for me with mentioned changes. Not that familiar with Buteo, so cannot tell much about logic changes.

File: msyncd/synchronizer.h (hunk 28 above)
Comment (renatofilho) [inline msyncd/synchronizer.h]: I can not include "glib.h" here because of conflicts with qt keywords 'signal/slot/etc..', to avoid include id I did a forward declaration of the struct, I can not do it with GSettings.

I can not use qt flag 'no_keywords' here because all code rely on these keywords.

**Comment by chriadam** [comment]
LGTM.  If nobody disagrees, I will merge this one next Monday.
